ecmascript/transforms/typescript/src/strip.rs

```diff
@@ -1113,26 +1113,50 @@ impl Strip {
 impl Visit for Strip {
     fn visit_ident(&mut self, n: &Ident, _: &dyn Node) {
         let is_type_only_export = self.is_type_only_export;
-        self.scope
-            .imported_idents
+        let entry = self
+            .scope
+            .referenced_idents
             .entry((n.sym.clone(), n.span.ctxt()))
-            .and_modify(|v| v.has_concrete |= !is_type_only_export);
+            .or_default();
+        entry.has_concrete |= !is_type_only_export;
 
         n.visit_children_with(self);
     }
 
-    fn visit_class_decl(&mut self, n: &ClassDecl, _: &dyn Node) {
-        n.visit_children_with(self);
-
-        self.decl_names.insert(n.ident.to_id());
-    }
-
     fn visit_decl(&mut self, n: &Decl, _: &dyn Node) {
         self.handle_decl(n);
 
         let old = self.non_top_level;
         self.non_top_level = true;
-        n.visit_children_with(self);
+        // Visit the declaration body and not the identifier. Otherwise
+        // type declarations could be unintentionally marked as concrete
+        // by the visit_ident function above.
+        match n {
```

Comment:
Is there a reason to handle this in `visit_decl` instead of `visit_class_decl`?
(It's not important, just curous)

Reply:
You mean create individual visitor functions for each declaration type? I could do that if it's preferable. Using `match` does verify that we're handling all types of declarations if one is ever added I guess. 🤷